launch_yaml/launch_yaml/parser.py

```diff
@@ -30,13 +31,22 @@ class Parser(frontend.Parser):
     @classmethod
     def load(
         cls,
-        stream: Union[str, io.TextIOBase],
+        file: Union[str, TextIO],
     ) -> (Entity, 'Parser'):
         """Return entity loaded from YAML file."""
-        if isinstance(stream, str):
-            stream = open(stream, 'r')
-        tree = yaml.safe_load(stream)
-        if len(tree) != 1:
-            raise RuntimeError('Expected only one root')
-        type_name = list(tree.keys())[0]
-        return (Entity(tree[type_name], type_name), cls())
+        try:
+            fileobj = open(file, 'r')
+            didopen = True
+        except TypeError:
+            fileobj = file
+            didopen = False
```

Comment:
@rotu I'm not very fond of using exceptions for flow control. Why not checking?

Reply:
Using exceptions for expected circumstances may not be natural in C++ but it's [idiomatic in Python](https://devblogs.microsoft.com/python/idiomatic-python-eafp-versus-lbyl/) so long as exceptions are specific and try-blocks are tight.

The reason here is because file can be a path-like object. Sure we're only expecting a `str` but it would be sensible to pass a `pathlib.Path` or `os.PathLike` and this code does the correct thing in all those cases.

Reply:
> The reason here is because file can be a path-like object. Sure we're only expecting a str but it would be sensible to pass a pathlib.Path or os.PathLike and this code does the correct thing in all those cases.

Consider updating type annotations if so.